CONTRIBUTION.md

```diff
@@ -0,0 +1,174 @@
+
+* Unix System
+* JDK 1.8
+* Maven 3.5 or later
+* Protocol Buffers 2.5
+* Internet connection for first build (to fetch all Maven and Hadoop dependencies)
+
+Additional requirements to run your first pseudo cluster:
+
+* docker
+* docker-compose
+
+Additional rquirements to execute different type of tests:
+
+* [Robot framework](https://robotframework.org/) (for executing acceptance tests)
+* docker-compose (to start pseudo cluster, also used for blockade and acceptance tests)
+* [blockade](https://pypi.org/project/blockade/) To execute network fault-injection testing.
+
+Optional dependencies:
+
+* [hugo](https://gohugo.io/) to include the documentation in the web ui.
+
+### Build the project
+
+The build is as simple as:
+
+```
+mvn clean install -DskipTests
+```
+
+And you can start our first cluster:
+
+```
+cd hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/ozone
+docker-compose up -d --scale datanode=3
+```
+
+### Helper scripts
+
+`hadoop-ozone/dev-support/checks` directory contains helper scripts to build and check your code. (Including findbugs and checkstyle). Use them if you don't know the exact maven gools / parameters.
+
+These scripts are executed by the CI servers, so it's always good to run them locally before creating a PR.
+
+### Maven build options:
+
+  * Use `-DskipShade` to exclude ozonefs jar file creation from the relase. It's way more faster, but you can't test Hadoop Compatible file system.
+  * Use `-DskipRecon` to exclude the Recon build (Web UI and monitoring) from the build. It saves about 2 additional minutes.
+  * Use `-Pdist` to build a distribution (Without this profile you won't have the final tar file)
+  * Use `-Pdocker-build` to build a docker image which includes Ozone
+  * Use `-Ddocker.image=repo/name` to define the name of your docker image
+  * USe `-Pdocker-push` to puseh the created docker image to the docker registry
+  
+## Contribute your modifications
+
+We  use github pull requests instead of uploading patches to JIRA. The main contribution workflow is the following:
+
+  1. Fork `apache/hadoop-ozone` github repository (first time)
+  2. Create a new Jira in HDDS project (eg. HDDS-1234)
+  3. Create a local branch for your contribution (eg. `git checkout -b HDDS-1234`)
+  4. Create your commits and push your branches to your personal fork.
+  5. Create a pull request on github UI 
+      * Please include the Jira link, problem description and testing instruction
+  6. Set the Jira to "Patch Available" state
+    
+## Code convention and tests
+
+We follow the code convention of Hadoop project (2 spaces instead of tabs, 80 char line width, ASF licence headers). The code checked with checkstyle, findbugs and various test frameworks.
+
+Please don't post / commit any code with any code violations (all checks are not checking the introduced violations as checks in Hadoop but all the available violations. One committed checkstyle issue will be reported in all of ht next pull requests.)
```

Comment:
Can we rewrite this `all checks are not checking the introduced violations as checks in Hadoop but all the available violations.` ?
May be `The in-built checks do not identify/isolate the new violations introduced by the patch. It will check and report all violations in the project (including your patch).